### PR TITLE
Deprecated idnssoaserial message

### PR DESCRIPTION
**Pull request by carma12**
A warning message (regarding the SOA serial deprecation) is shown on the WebUI and CLI every time a new DNS zone is added (even if the '--serial' option is not explicitly set) or the SOA serial is modified.

This should be managed by setting the 'idnssoaserial' as deprecated and not required parameter.

Also, on the WebUI, the SOA serial textbox must be disabled (non-editable) to prevent the 'ValidationError' message from being shown when this specific field is manually set:
image

Fixes: https://pagure.io/freeipa/issue/9249
Signed-off-by: Carla Martinez <carlmart@redhat.com>

### CODE REVIEW

**Comment by carma12** [comment]
The failure in test_commands is a new issue ([9244](https://pagure.io/freeipa/issue/9244) Nightly test failure in test_commands.py::TestIPACommand::test_ipa_cacert_manage_prune). A tentative fix is available in PR https://github.com/freeipa/freeipa/pull/6466.

File: ipaserver/plugins/dns.py [hunk 1]
```diff
@@ -2431,15 +2431,14 @@ class dnszone(DNSZoneBase):
...
+        Int('idnssoaserial?',
             # Deprecated
             cli_name='serial',
             label=_('SOA serial'),
             doc=_('SOA record serial number'),
             minvalue=1,
             maxvalue=4294967295,
-            default_from=_create_zone_serial,
-            autofill=True,
+            deprecated=True,
```
Comment (rcritten) [inline ipaserver/plugins/dns.py:2441]: This not only deprecates the option but makes it not possible for a user to provide it, either on the CLI or the UI. Is that intentional?

**Comment by rcritten** [comment]
Can you add the pagure ticket # to the commit(s)?

**Comment by carma12** [comment]
> Can you add the pagure ticket # to the commit(s)?

As far as I am aware, there is no Pagure ticket for this bug, only the [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=2108630) one.

File: ipaserver/plugins/dns.py (hunk 1 above)
Comment (carma12) [reply ipaserver/plugins/dns.py:2441]: Yes, that is the main point behind this fix (aside from not showing the warning message anymore). 

As the SOA serial is managed automatically via autoincrement, the idea is that the user couldn't provide the SOA serial parameter manually (via CLI or WebUI). Also, we prevent it from being generated via autofill, which naturally avoids the warning.

Comment (rjeffman) [reply ipaserver/plugins/dns.py:2441]: As for the CLI, the option was already not available for the user to set SOA serial. The issue was that the API was still accepting it.

**Comment by rcritten** [comment]
@carma12 you'll need to clone a BZ to pagure. That is what we typically do to keep a history of changes. I can help if needed.

**Comment by carma12** [comment]
Hi @rcritten 
I have already cloned the BZ and added the generated Pagure ticket to all the commits.

**Comment by flo-renaud** [comment]
Before the patch, we had the following behavior:
```
# ipa dnszone-add zone.test --serial 1
Usage: ipa [global-options] dnszone-add NAME [options]

ipa: error: no such option: --serial
# ipa dnszone-add zone.test.
ipa: WARNING: 'idnssoaserial' option is deprecated. Value will be ignored.
  Zone name: zone.test.
...
# ipa dnszone-add zone2.test. --setattr idnssoaserial=1
ipa: WARNING: 'idnssoaserial' option is deprecated. Value will be ignored.
...
  SOA serial: 1
...
```
With the patch:
```
# ipa dnszone-add zone.test --serial 1
Usage: ipa [global-options] dnszone-add NAME [options]

ipa: error: no such option: --serial
# ipa dnszone-add zone.test.
  Zone name: zone.test.
...
# ipa dnszone-add zone2.test. --setattr idnssoaserial=1
ipa: ERROR: invalid 'idnssoaserial': this option is deprecated
#
```
=> the unexpected warning is not there any more and it is not possible to directly set the serial. Same thing with dnszone-mod.

The UI displays the soa serial as read-only, which is the expectation, and updates to other fields do not trigger the unwanted warning but update the serial.
LGTM.

**Comment by carma12** [comment]
master:

* c74c701cacc9294150059c512d6b9c9b8c16f46b Set 'idnssoaserial' to deprecated
* b326b4afae31ff7b2f21e52badf3c450746b77b7 ipatest: Remove warning message for 'idnssoaserial'
* a3b4b476b9e8e9a715694a06369d63c2706e47db webui: Set 'SOA serial' field as read-only
* 55ef0008b80717b7dabb572aef488c9826a5b3f5 Update API and VERSION

**Pull request closed without merging**